packages/ui/src/app/pages/Settings/Settings.tsx

```diff
@@ -36,13 +41,108 @@ export const Settings = () => {
     { title: t('network'), active: currentTab === 'SETTINGS', onClick: () => setCurrentTab('SETTINGS') },
     //{ title: t('language'), active: currentTab === 'LANGUAGE', onClick: () => setCurrentTab('LANGUAGE') },
   ]
+
+  const form = useForm()
+  const [customFaucetEndpoint, customRpcEndpoint, customQueryEndpoint] = form.watch([
+    'settings.customFaucetEndpoint',
+    'settings.customRpcEndpoint',
+    'settings.customQueryEndpoint',
+  ])
+  const [, storeCustomEndpoints] = useLocalStorage<NetworkEndpoints>('custom_endpoint')
+  const [isValidFaucetEndpoint, setIsValidFaucetEndpoint] = useState(true)
+  const [isValidRpcEndpoint, setIsValidRpcEndpoint] = useState(true)
+  const [isValidQueryEndpoint, setIsValidQueryEndpoint] = useState(true)
+  const [customSaveStatus, setCustomSaveStatus] = useState<'Init' | 'Saving' | 'Done'>('Init')
+
   const switchNetwork = (network: NetworkType | null) => {
     if (network) {
       setNetwork(network)
       window.location.reload()
     }
   }
 
+  useEffect(() => {
+    if (network === 'custom') {
+      form.setValue('settings.customRpcEndpoint', endpoints.nodeRpcEndpoint)
+      form.setValue('settings.customQueryEndpoint', endpoints.queryNodeEndpoint)
+      form.setValue('settings.customFaucetEndpoint', endpoints.membershipFaucetEndpoint)
+    }
+  }, [network, endpoints])
+
+  const saveSettings = () => {
+    if (
+      /^(http|https):\/\//i.test(customFaucetEndpoint) === false ||
+      /^(ws|wss):\/\//i.test(customRpcEndpoint) === false ||
+      /^(http|https):\/\//i.test(customQueryEndpoint) === false
+    ) {
+      return
+    }
+
+    setCustomSaveStatus('Saving')
+
+    const _update = async () => {
+      let needReload = true
+
+      // check faucet endpoint
+      try {
+        const faucetStatusEndpoint = customFaucetEndpoint.replace(new RegExp('register$'), 'status')
+        const response = await fetch(faucetStatusEndpoint)
+        await response.json()
+        setIsValidFaucetEndpoint(true)
+      } catch (err) {
+        setIsValidFaucetEndpoint(false)
+        needReload = false
+      }
+
+      // check GraphQL endpoint
+      try {
+        const query = `
+          query {
+            electedCouncils{
+              id
+            }
+          }`
+        const response = await fetch(customQueryEndpoint, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({ query }),
+        })
+        await response.json()
+        setIsValidQueryEndpoint(true)
+      } catch (err) {
+        setIsValidQueryEndpoint(false)
+        needReload = false
+      }
+
+      // check RPC endpoint
+      try {
+        const wsProvider = new WsProvider(customRpcEndpoint, false, undefined, 4000)
+        await wsProvider.connect()
```

Comment:
This doesn't work for me I can save with any url starting with `wss://...` even if it doesn't exist.
Could you check if one of these works please?
```suggestion
        const wsProvider = new WsProvider(customRpcEndpoint)
        await wsProvider.isConnected
```
```suggestion
        const wsProvider = new WsProvider(customRpcEndpoint)
        await new Promise((resolve, reject) => {
          wsProvider.on('connected', resolve)
          wsProvider.on('error', reject)
        })
```
Also I think 4 seconds is way too long especially since the validation is `true` in the meantime.